subsys/bluetooth/host/settings.c

```diff
@@ -324,21 +374,23 @@ __weak void bt_testing_settings_delete_hook(const char *key)
 int bt_settings_store(const char *key, uint8_t id, const bt_addr_le_t *addr, const void *value,
 		      size_t val_len)
 {
-	int err;
-	char id_str[4];
 	char key_str[BT_SETTINGS_KEY_MAX];
+	char id_str[4];
+	int err;
 
 	if (addr) {
 		if (id) {
 			u8_to_dec(id_str, sizeof(id_str), id);
 		}
 
-		bt_settings_encode_key(key_str, sizeof(key_str), key, addr, (id ? id_str : NULL));
+		err = bt_settings_encode_key(key_str, sizeof(key_str), key, addr,
+					     (id ? id_str : NULL));
 	} else {
-		err = snprintk(key_str, sizeof(key_str), "bt/%s", key);
-		if (err < 0) {
-			return -EINVAL;
-		}
+		err = bt_settings_encode_key_no_addr(key_str, sizeof(key_str), key);
```

Comment:
I'm OK with 2 functions, but an alternative could also be to just have a single function that takes an `addr` which could be NULL and handle it there

Reply:
I did push a draft in this PR with both implementation, and the code size with ~ 20 bytes more due to added branching instructions to check for NULL, and to keep old behavior of optional parameters and the string output on truncation etc.

Reply:
Interesting. You'd have the same amount of branching, so not sure why it would be 20 bytes more due to that